platform/ios/src/MGLMapView.mm

```diff
@@ -2241,6 +2244,28 @@ - (CLLocationDegrees)angleBetweenPoints:(CGPoint)west east:(CGPoint)east
     return degrees;
 }
 
+- (CGFloat)newDegreesForRotationGesture:(UIRotationGestureRecognizer *)rotate {
+    CGFloat newDegrees;
+
+    if (self.isZooming) {
+        CGFloat rotation = rotate.rotation;
+        CGFloat newRotation = 0;
+
+        CGFloat b = MGLRadiansFromDegrees(self.rotationThresholdWhileZooming * 2);
+        CGFloat a = MGLRadiansFromDegrees(self.rotationThresholdWhileZooming);
+        if ((rotation >= b) || (rotation <= -b)) {
+            newRotation = rotation;
+        }
+        else if (rotation >= a) {
+            newRotation = -b*0.5*(cos(M_PI*(rotation-a)/(b-a)) - 1.0);
```

Comment:
Please add comments that explain what these are doing and/or break these kinds of algorithms out into more digestible parts that are self-descriptive.